internal/scheduleringester/schedulerdb.go

```diff
@@ -37,7 +37,10 @@ func (s *SchedulerDb) Store(ctx context.Context, instructions *DbOperationsWithM
 			shouldRetry := armadaerrors.IsNetworkError(err) || armadaerrors.IsRetryablePostgresError(err)
 			return shouldRetry, err
 		}, s.initialBackOff, s.maxBackOff)
-		multierror.Append(result, err)
+		multiAppendErr := multierror.Append(result, err)
```

Comment:
This should be `multierror.Append(result, err)`. Don't check if the error isn't `nil` here. By doing so the function would return prematurely at the first error. We want to collect all errors before returning.
`multierror.Append` works like the regular `append` for slices in that it adds information to an existing error. But we need to update the variable in-place (like with the regular `append`), which we weren't doing.